packages/utils/src/cmds/transform_translate.js

```diff
@@ -0,0 +1,381 @@
+ * d2-utils transform_translate \
+ *   --in-dir ~/development/dhis2/maintenance-app/src/i18n \
+ *   --out-dir ~/development/dhis2/project-doom/i18n \
+ *   --override-existing-files \
+ *   --app-name maintenance-app \
+ *   --languages fr,ar \
+ *   --delete-old-files \
+ *   --log-missing-keys
+ */
+const path = require('path')
+const fs = require('fs')
+const log = require('@dhis2/cli-helpers-engine').reporter
+
+const CONSUMING_ROOT = path.join(process.cwd())
+const TRANSLATION_IN_DIR = path.join(CONSUMING_ROOT, 'src/i18n')
+const TRANSLATION_OUT_DIR = path.join(CONSUMING_ROOT, 'i18n')
+
+const fileIsOldTranslationFile = fileName => fileName.match(/\.properties$/)
+
+const appNamePattern = new RegExp('[a-z]+(-[a-z]+)?-app')
+
+/**
+ * \u[a-zA-Z0-9] will be interpreted as unicode characters and need to be escaped.
+ * The escaped sequence will look like this: %5Cu[a-zA-Z0-9]
+ *
+ * The translation needs to be split by whitespaces first in order to create the
+ * correct structure of the new translation
+ */
+const splitTranslation = (originalTranslation, translation) => {
+    const newParts = translation.split(' ').reduce((parts, curSplit) => {
+        const latestPart = parts[parts.length - 1]
+        const latestPartEscaped = escape(latestPart)
+        const curSplitEscaped = escape(curSplit)
+
+        if (
+            parts.length > 0 &&
+            latestPartEscaped.length + curSplitEscaped.length < 77
+        ) {
+            parts[parts.length - 1] += ` ${curSplit}`
+            return parts
+        }
+
+        if (curSplitEscaped.length < 77) {
+            parts.push(curSplit)
+            return parts
+        }
+
+        curSplitEscaped.match(/.{1,76}(?=(%5Cu))?/g).forEach(escapedSplit => {
+            parts.push(unescape(escapedSplit))
+        })
+
+        return parts
+    }, [])
+
+    return newParts
+}
+
+const getTemplateMainLanguage = () => `
+msgid ""
+msgstr ""
+"Project-Id-Version: i18next-conv\\n"
+"MIME-Version: 1.0\\n"
+"Content-Type: text/plain; charset=utf-8\\n"
+"Content-Transfer-Encoding: 8bit\\n"
+"Plural-Forms: nplurals=2; plural=(n != 1)\\n"
+"POT-Creation-Date: 2019-08-13T13:38:22.134Z\\n"
+"PO-Revision-Date: 2019-08-13T13:38:22.134Z\\n"
+`
+
+const getTemplateAlternativeLanguage = (appName, language) => `
+msgid ""
+msgstr ""
+"Project-Id-Version: PACKAGE VERSION\\n"
+"Report-Msgid-Bugs-To: \\n"
+"POT-Creation-Date: 2018-04-10 19:51+0000\\n"
+"PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\\n"
+"Last-Translator: FULL NAME <EMAIL@ADDRESS>\\n"
+"Language-Team: LANGUAGE <LL@li.org>\\n"
+"Language: ${language}\\n"
+"MIME-Version: 1.0\\n"
+"Content-Type: text/plain; charset=UTF-8\\n"
+"Content-Transfer-Encoding: 8bit\\n"
+"X-Generator: Translate Toolkit 2.2.5\\n"
+"X-Pootle-Path: /${language}/${appName}/${language}.po\\n"
+"X-Pootle-Revision: 29879\n"
+`
+
+exports.command = 'transform_translate'
+exports.describe = 'Transform old translation file style to new style'
+exports.builder = {
+    appName: {
```

Comment:
I think it makes sense to have some arguments, rather than having everything as options.  `appName`, `inDir`, and `outDir` seem like good candidates.  You can specify positional arguments (required and optional) in the `command` name like this: `transform_translate <appName> [inDir] [outDir]`, though it might make sense that both dirs be required and I'm not sure it's obvious to the user why `appName` should be...

Reply:
@amcgee How do I use options when using the nested namespaces approach? 
Now the command is a key, not a string anymore

Reply:
@Mohammer5 not sure I understand the question... I am able to see the help and pass options properly with `./bin/d2-app i18n --help`, `./bin/d2-app i18n modernize --help`, and `./bin/d2-app i18n modernize --inDir <something>` on this branch.  What do you mean by "the command is a key, not a string anymore"?

I also restructured the directories to match the structure we use elsewhere in #126 (feel free to merge that or just reference it) - keeping helper code outside the `commands` directory and having a separate file for each namespace (`i18n`) and command (`modernize`)

Reply:
Ah, my bad I see what you mean - when using the `commands` object to create the namespace the key becomes the command name.  I believe you can still pass the `command: 'modernize <appName> [inDir] [outDiir]'` in that case by placing them in the key or in the `command` property of the command object, or just avoid using `commands: { ... }` and use `builder: yargs => yargs.command(modernize)` instead

Reply:
Examples, tested and these all work:

```js
const modernize = {
    describe: 'Transform old translation file style to new style',
    builder,
    handler,
}

module.exports = namespace('i18n', {
    description: 'Handle translations in apps',
    commands: { 'modernize <appName>': modernize },
})
```

```js
const modernize = {
    command: 'modernize <appName>'
    describe: 'Transform old translation file style to new style',
    builder,
    handler,
}

module.exports = namespace('i18n', {
    description: 'Handle translations in apps',
    builder: yargs => yargs.command(modernize)
})
```

```js
//src/commands/i18n/modernize.js
module.exports = {
    command: 'modernize <appName>'
    describe: 'Transform old translation file style to new style',
    builder,
    handler,
}

// src/commands/i18n.js
module.exports = namespace('i18n', {
    description: 'Handle translations in apps',
    builder: yargs => yargs.commandDir('i18n')
})
```